jobs/sql/extractions/functions/temp_program.sql

```diff
@@ -388,4 +389,19 @@ BEGIN
       and       (_encounter_type is null or o.encounter_type_id = _encounter_type);
     RETURN ret;
 END
-#
\ No newline at end of file
+#
+
+DROP FUNCTION IF EXISTS temp_program_earliest_patient_state_date;
+#
+CREATE FUNCTION temp_program_earliest_patient_state_date(_patient_program_id int, _patient_state_id int(11))
+    RETURNS date
+    DETERMINISTIC
+BEGIN
+    DECLARE ret date;
+	select start_date into ret from patient_state ps 
+	where patient_program_id = _patient_program_id
+	and ps.state = _patient_state_id
+	order by start_date desc limit 1;    
```

Comment:
Won't this actually return the _latest_ patient state date?  The function name says "earliest".